src/core/public/chrome/ui/loading_indicator.tsx

```diff
@@ -39,16 +39,26 @@ export class LoadingIndicator extends React.Component<LoadingIndicatorProps, { v
     visible: false,
   };
 
+  private timer: any;
+  private increment = 1;
+
   componentDidMount() {
     this.loadingCountSubscription = this.props.loadingCount$.subscribe((count) => {
-      this.setState({
-        visible: count > 0,
-      });
+      if (this.increment > 1) {
+        clearTimeout(this.timer);
+      }
+      this.increment += this.increment;
+      this.timer = setTimeout(() => {
+        this.setState({
+          visible: count > 0,
+        });
+      }, 250);
```

Comment:
So, the delay is reset every time `loadingCount` emits a new value, meaning that if there are consecutive requests firing at an interval < `250`, the loader will never be displayed until 250ms after the last one. Is that the expected behavior?

Shouldn't we start the timer the first time `loadingCount$` is greater than 0, but not on the next emissions until  `loadingCount` is back to `0`?

Reply:
I'll be honest, if you have experience with this code, feel free to commit on top of it. The way the subscription numbers shifted felt weird.

In any case the problem I was trying to solve from a UI perspective with the current functionality:

The loading action "blips" for minor events. Basically, ANY time the subscription changed (which is often) it would signal. Less than a second of loading, and the user won't even notice the load, but they would notice an annoying animation on the page. We'd prefer not to show it in those cases, and only show it for sustained loads. I _think_ what your'e saying is the correct way to handle it. Look for it to go back to zero, then start the count, and see if the number is larger than zero.

Reply:
+1 to Pierre's comments

I can add this to my queue if you're not sure how to implement it but I don't think I'll get to it for at least a few days (if someone else has time before me, feel free to jump in)

Reply:
I'm gonna give this a shot tonight. I'll bug you all tomorrow for a review.